data/crac/crac-api/src/main/java/com/powsybl/openrao/data/cracapi/Crac.java

```diff
@@ -542,44 +544,73 @@ default State getState(String contingencyId, Instant instant) {
     RaUsageLimitsAdder newRaUsageLimits(String instantName);
 
     /**
-     * Import CRAC from a file
+     * Get the CRAC format
+     * @param filename               CRAC file name
+     * @param inputStream            CRAC data
+     * @return the CRAC format (if found)
+     */
+    static String getCracFormat(String filename, InputStream inputStream) throws IOException {
+        byte[] bytes = getBytesFromInputStream(inputStream);
+        return new ServiceLoaderCache<>(Importer.class).getServices().stream()
+            .filter(importer -> importer.exists(filename, new ByteArrayInputStream(bytes)))
+            .findAny()
+            .orElseThrow(() -> new OpenRaoException("No suitable CRAC importer found."))
```

Comment:
Maybe you can mutualize this code for this method and the following one (readWithContext).